lean-client-js-core/src/commands.ts

```diff
@@ -139,6 +139,31 @@ export interface InfoResponse extends CommandResponse {
     record?: InfoRecord;
 }
 
+export interface SymbolRequest extends Request {
+    command: 'symbols';
+    file_name: string;
+}
+
+export declare type SymbolKind
+    = 'constructor'
+    | 'field'
+    | 'class'
+    | 'inductive'
+    | 'instance'
+    | 'def'
+    | 'theorem'
```

Comment:
One question that came up in discussion with @Julian; where should the translation from lean concepts to VScode/LSP concepts go; in the lean server, or in the extension?

Reply:
The motivation for the leading question was this is going to immediately need to be copied in `lean-language-server` if it doesn't go further upstream than that, but there's presumably 3 choices -- server, extension (and then copy pasted), or in some JS library function here.

Reply:
I'd keep the server simple if possible.  Having the raw data could also be useful for other applications.  Not that I expect a lot of new development in Lean 3 though.

In this case, copying the vscode implementation into `lean-language-server` would be my preferred solution.